valhalla/meili/routing.h

```diff
@@ -305,8 +305,14 @@ find_shortest_path(baldr::GraphReader& reader,
                    const float max_time);
 
 // Route path iterator. Methods to assist recovering route paths from Labels.
-class RoutePathIterator : public std::iterator<std::forward_iterator_tag, const Label> {
+class RoutePathIterator {
 public:
+  using iterator_category = std::forward_iterator_tag;
+  using value_type = const Label;
+  using difference_type = const std::ptrdiff_t;
+  using pointer = const Label*;
+  using reference = const Label&;
```

Comment:
basically what `std::iterator` was doing under the hood (also the only thing it was really doing).

Reply:
let's reason over the constness together..